src/components/transport_manager/src/transport_manager_impl.cc

```diff
@@ -673,21 +712,203 @@ TransportManagerImpl::ConnectionInternal* TransportManagerImpl::GetConnection(
   return NULL;
 }
 
-void TransportManagerImpl::OnDeviceListUpdated(TransportAdapter* ta) {
-  LOG4CXX_TRACE(logger_, "enter. TransportAdapter: " << ta);
-  const DeviceList device_list = ta->GetDeviceList();
-  LOG4CXX_DEBUG(logger_, "DEVICE_LIST_UPDATED " << device_list.size());
-  for (DeviceList::const_iterator it = device_list.begin();
-       it != device_list.end();
+TransportManagerImpl::ConnectionInternal*
+TransportManagerImpl::GetActiveConnection(
+    const DeviceUID& device, const ApplicationHandle& application) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  LOG4CXX_DEBUG(logger_,
+                "DeviceUID: " << device
+                              << " ApplicationHandle: " << application);
+  for (std::vector<ConnectionInternal>::iterator it = connections_.begin();
+       it != connections_.end();
+       ++it) {
+    if (it->device == device && it->application == application && it->active_) {
+      LOG4CXX_DEBUG(logger_, "ConnectionInternal. It's address: " << &*it);
+      return &*it;
+    }
+  }
+  return NULL;
+}
+
+namespace {
+
+struct TAFinder {
```

Comment:
Seems like an poorly-named struct, since it is looks specifically for `IOS_BT` connected devices. Maybe `BTAdapterFinder`?

Reply:
@jacobkeeler Fixed in b969449ecb89eb95e288d9d0024d1b1d2d1f66f0